src/appsettings/settings.py

```diff
@@ -916,6 +998,20 @@ def __init__(self, name="", default=list, *args, **kwargs):
         """
         super(ListSetting, self).__init__(name=name, default=default, *args, **kwargs)
 
+    def decode_environ(self, value):
+        """
+        Decode JSON value or split value by a delimiter to a list, if JSONDecodeError is raised.
+
+        Default delimiter is a colon, can be changed via attribute ``delimiter``.
+
+        Args:
+            value (str):
+
+        Returns:
+            list:
+        """
+        return list(super(ListSetting, self).decode_environ(value))
```

Comment:
I believe this may turn scalar value into a list, which is not expected.

Reply:
I am not sure what do you mean? Like value could be (for example) legal JSON boolean value and this would produce a list with one boolean value inside?

How to solve it? Check `isinstance(e, Iterable)`? The parent `IterableSetting.decode_environ` should always return a list (splitting returns a list and the JSON should too), so we may check this?

Reply:
> Like value could be (for example) legal JSON boolean value and this would produce a list with one boolean value inside?

Indeed, but as I tried it, the main problem is with strings and there is probably no reasonable way to handle that, so I just retract my comment.